src/ctia/bulk/core.clj

```diff
@@ -35,28 +38,39 @@
   [k]
   (get inverted-bulk-entity-mapping k))
 
-(defn create-fn
+(s/defn create-fn
   "return the create function provided an entity type key"
-  [k auth-identity params]
+  [k auth-identity params {{:keys [write-store]} :StoreService :as _services_} :- APIHandlerServices]
   #(write-store
     k store/create-record
     % (auth/ident->map auth-identity) params))
 
-(defn read-fn
+(s/defschema ReadFnServices
+  {:StoreService (-> APIHandlerServices
+                     (st/get-in [:StoreService])
+                     (st/select-keys [:read-store])
+                     (st/assoc s/Keyword s/Any))
+   s/Keyword s/Any})
```

Comment:
perhaps [st/open-schema](https://cljdoc.org/d/metosin/schema-tools/0.12.2/api/schema-tools.core#open-schema)?

Reply:
Thanks, but I don't like its implementation (it recursively opens all maps).